rocks-db/src/columns/asset.rs

```diff
@@ -2073,12 +2071,18 @@ pub fn merge_complete_details_fb_simple_raw<'a>(
                                 owner_pubkey = merged_owner.pubkey;
                                 owner_owner = merged_owner.owner;
                                 owner_delegate = merged_owner.delegate;
-                                owner_owner_type = merged_owner.owner_type;
                                 owner_owner_delegate_seq = merged_owner.owner_delegate_seq;
                                 owner_is_current_owner = merged_owner.is_current_owner;
                                 other_known_owners.remove(new_owner_pubkey.bytes());
                             }
                         }
+                        // Since the owner type can only be changed through a mint account state update,
+                        // we should avoid modifying it every time we receive token account updates for the corresponding mint.
```

Comment:
won't there be an issue when solving the fork if the owner_type on the alternative chain is different?

Reply:
it's hard to imagine such a scenario because owner_type is changing only when token supply is changing, and it's not happening that often as token transfers, and it will be changed only when supply is changing from `1` to `>1`. also we don't have yet any proofs that forks affect accounts same as transactions. and if it is then we should rethink merge logic for all the fields, not only owner_type, because at the moment we simply merge everything.

that's a really good point, thanks for it. but I think we need to have proofs first to understand what exactly happening there and than refactor code if it needs.